luacompiler/src/lib/irgen/register_map.rs

```diff
@@ -129,6 +129,15 @@ impl<'a> RegisterMap<'a> {
         None
     }
 
+    pub fn is_local(&self, name: &str) -> bool {
+        for map in self.reg_maps.iter().rev() {
```

Comment:
Why reverse this? I think this function is (roughly) equivalent to something like `self.reg_maps.iter().find(|x| x.contains_key(name))`.